Gems/ROS2/Code/Source/Camera/ROS2CameraSensorComponent.cpp

```diff
@@ -42,47 +33,54 @@ namespace ROS2
         auto* serialize = azrtti_cast<AZ::SerializeContext*>(context);
         if (serialize)
         {
-            serialize->Class<ROS2CameraSensorComponent, ROS2SensorComponent>()
-                ->Version(3)
-                ->Field("VerticalFieldOfViewDeg", &ROS2CameraSensorComponent::m_verticalFieldOfViewDeg)
-                ->Field("Width", &ROS2CameraSensorComponent::m_width)
-                ->Field("Height", &ROS2CameraSensorComponent::m_height)
-                ->Field("Depth", &ROS2CameraSensorComponent::m_depthCamera)
-                ->Field("Color", &ROS2CameraSensorComponent::m_colorCamera);
+            serialize->Class<ROS2CameraSensorComponent, ROS2SensorComponent>()->Version(4)->Field(
+                "CameraSensorConfig", &ROS2CameraSensorComponent::m_cameraConfiguration);
         }
     }
 
-    void ROS2CameraSensorComponent::Activate()
+    void ROS2CameraSensorComponent::AddImageSource(const Camera::ImagePostProcessingFn& imagePostProceessingFn, bool depth)
```

Comment:
I do not like this refactor, it makes code overconstrained to use case. We can have in future other sensors (e.g. optical flow). Maybe use here template argument?
```template typename T void ROS2CameraSensorComponent::AddImageSource<T>(const Camera::ImagePostProcessingFn& imagePostProceessingFn)```
This pattern is used are already with the helper `CreatePair<T>`. Code will follow DRY, won't be constrained and without ifs.